### ISSUE TITLE
Distance Field Font renders certain characters with improper padding

### ISSUE THREAD

**Issue by SasLuca** [libgdx/libgdx#4106]
#### Issue details

When rendering text on screen using Distance Field Fonts as described in the tutorial, some letters have the wrong padding. I tried this with multiple fonts and they all had this problem. The last one I tried was Arial Black with the same settings in Hiero as described in this tutorial: [https://github.com/libgdx/libgdx/wiki/Distance-field-fonts](https://github.com/libgdx/libgdx/wiki/Distance-field-fonts)

I noticed the problem always seemed to be with the same letters, like with the letter 'j' in this case 
![6e03375d772b45e243484ccb470a10e2](https://cloud.githubusercontent.com/assets/15983269/15627771/4e6147c8-24f7-11e6-9ce0-957ba645e51f.png)

As you can see, the first 'i' after the 'j' is too close. I tried this again with another font ("Roboto") and it still had the same problem with the same settings. Note that this problem persisted regardless of my settings or my font.
![524f56d475893268c3385e2572cda93e](https://cloud.githubusercontent.com/assets/15983269/15627781/aa5fc77a-24f7-11e6-9c91-fba943575bd5.png)
#### Reproduction steps/code

Here is the code I used. This code was compiled using IntelliJ and Gradle, with LibGDX version 1.6.2 and 1.9.3, the result was the same regardless of the LibGDX version.

``` java
package com.mygdx.game;

import com.badlogic.gdx.ApplicationAdapter;
import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.graphics.Color;
import com.badlogic.gdx.graphics.GL20;
import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.graphics.g2d.BitmapFont;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.badlogic.gdx.graphics.g2d.TextureRegion;
import com.badlogic.gdx.graphics.glutils.ShaderProgram;

public class MyGdxGame extends ApplicationAdapter
{
    SpriteBatch batch;
    BitmapFont font;
    ShaderProgram fontShader;

    @Override public void create ()
    {
        Texture text = new Texture(Gdx.files.internal("test.png"));
        text.setFilter(Texture.TextureFilter.Linear, Texture.TextureFilter.Linear);

        batch = new SpriteBatch();
        font = new BitmapFont(Gdx.files.internal("test" + ".fnt"), new TextureRegion(text), false);
        font.setColor(Color.BLACK);
        font.getData().setScale(3);
        fontShader = new ShaderProgram(Gdx.files.internal("font.vert"), Gdx.files.internal("font.frag"));
    }

    @Override
    public void render ()
    {
        Gdx.gl.glClearColor(1, 1, 1, 1);
        Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT);
        batch.setShader(fontShader);
        batch.begin();

        font.draw(batch, "jii", 0, 100);
        batch.end();
    }
}
```
#### Please select the affected platforms
- [X] Android
- [X] iOS
- [X] HTML/GWT
- [X] Windows
- [X] Linux
- [X] MacOS

**Comment by ThomasWrobel** [comment]
Does this look ok with other shaders? I would assume the problem is in how
the fonts texture map has arranged the letters OR in how that texture is
used to create a bitmap for the shader.
I dont think it will be a issue in the distance field shader itself.
Have you tried looking at the test.png itself ? Its possible even with the
gradient style for the distance field you can still see if the spacing
looks off on some letters.


--
http://lostagain.nl <-- our company site.
http://fanficmaker.com <-- our, really,really, bad story generator.

**Comment by SasLuca** [comment]
@ThomasWrobel I never tried using it with different shaders. I just used the shader mentioned in the tutorial. I also tried setting the smoothing using 2 uniform variables for scale and spread but it doesn't change anything. Also, there is no problem with the padding in the test.png. You can view it here: [https://gyazo.com/09c8b19db8d25a514bba1e134d67de4a](https://gyazo.com/09c8b19db8d25a514bba1e134d67de4a)

**Comment by xoppa** [comment]
Possible duplicate of or related to #4106,

**Comment by ruffiely** [comment]
Have the sam problem any news on it?

**Comment by badlogic** [comment]
Potential fix here, but that messes with the bmfont file format padding order https://github.com/libgdx/libgdx/issues/4282

**Comment by NathanSweet** [comment]
Looks fixed by #4282, if not will reopen.
